File: java/pe001.java
//file: pe001.java

public class pe001 {
  public static void main( String[] args ) {
    int limit = 1000;
    int sum = 0;
    for (int i = 0; i < limit; i++) {
      if (((i % 5) == 0) || ((i % 3) == 0)) {
        sum += i;
      }
    }
    System.out.println(sum);
  }
}

File: ocaml/pe001.ml
let divisible_5 x = x mod 5 = 0 ;;
let divisible_3 x = x mod 3 = 0 ;;

let rec pe001 count sum =
  if count > 0 then
    if (divisible_5 count || divisible_3 count) then
      pe001 (count - 1) (sum + count)
    else 
      pe001 (count - 1) sum
  else 
    Printf.printf "%d\n" sum ;;

let main () =
  pe001 999 0

let () = main ()
